src/Tools/ExternalAccess/Razor/Remote/JsonSerializableRazorPinnedSolutionInfoWrapper.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.Razor
+{
+    /// <summary>
+    /// A wrapper for a solution that can be used by Razor for OOP services that communicate via System.Text.Json
+    /// </summary>
+    internal readonly record struct JsonSerializableRazorPinnedSolutionInfoWrapper(
+        [property: JsonPropertyName("data1")] long Data1,
+        [property: JsonPropertyName("data2")] long Data2)
```

Comment:
interesting.  would like to know more about this.  when are you guuys pinning solutions?

Reply:
> interesting. would like to know more about this. when are you guuys pinning solutions?

hopefully never? This is just a version of [RazorPinnedSolutionInfoWrapper](https://github.com/davidwengier/roslyn/blob/06d3f153ed6af6f2b78028a1e1e6ecc55c8ff101/src/Tools/ExternalAccess/Razor/Remote/RazorPinnedSolutionInfoWrapper.cs#L11), which is really just [Checksum](https://github.com/davidwengier/roslyn/blob/main/src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs#L24), but this serializes with STJ and that uses messagepack. I could add STJ attributes to `RazorPinnedSolutionInfoWrapper`, but then I'd need them on `Checksum` too.

Reply:
> @eiriktsarpalis how do longs get serialized in STJ? Are they guaranteed to roundtrip?

They serialize to JSON numbers which have arbitrary precision. They are guaranteed to roundtrip.

Reply:
Thanks!